lib/api.js

```diff
@@ -115,6 +115,20 @@ module.exports = function (PersistObjectTemplate, baseClassForPersist) {
         }
     }
     PersistObjectTemplate._injectTemplateFunctions = function (template) {
+        function logExceptionAndRethrow(exception, logger, template, query, activity) {
```

Comment:
you don't need to change this at the time since I know we have to test this, but I'd prefer if we didn't scope functions within functions if they're named. We should move them outside the scope of the function.